src/seml/project.py

```diff
@@ -48,84 +59,121 @@ def init_project(
             type=bool,
         ):
             exit(1)
-    logging.info(f'Initializing project in "{directory}" using template "{template}"')
-
-    template_path = (
-        Path(str(importlib.resources.files('seml')))
-        / 'templates'
-        / 'project'
-        / template
-    )
-    if not template_path.exists():
-        logging.error(f'Template "{template}" does not exist')
-        exit(1)
-
-    if project_name is None:
-        project_name = directory.name
-    if user_name is None:
-        user_name = os.getenv('USER', os.getenv('USERNAME', 'user'))
-    if user_mail is None:
-        user_mail = 'my@mail.com'
-    format_map = dict(
-        project_name=project_name, user_name=user_name, user_mail=user_mail
-    )
-
-    gitignore_path = template_path / '.gitignore'
-    if gitignore_path.exists():
-        ignore_file = parse_gitignore(gitignore_path)
-    else:
-
-        def ignore_file(file_path: str):
-            return False
-
-    # Copy files one-by-one
-    for src in template_path.glob('**/*'):
-        # skip files ignored by .gitignore
-        if ignore_file(str(src)):
-            continue
-        # construct destination
-        file_name = src.relative_to(template_path)
-        target_file_name = Path(str(file_name).format_map(format_map))
-        dst = directory / target_file_name
-        # Create directories
-        if src.is_dir():
-            if not dst.exists():
-                dst.mkdir()
-        elif not dst.exists():
-            # For templates fill in variables
-            if src.suffix.endswith('.template'):
-                dst = dst.with_suffix(src.suffix.removesuffix('.template'))
-                dst.write_text(src.read_text().format_map(format_map))
-            else:
-                # Other files copy directly
-                dst.write_bytes(src.read_bytes())
+
+    tmp_dir = checkout_template_repo(git_remote, git_commit)
+    with checkout_template_repo(git_remote, git_commit) as tmp_dir:
+        template_path = tmp_dir / 'templates' / template
+        if not template_path.exists():
+            logging.error(f'Template "{template}" does not exist')
+            exit(1)
+
+        logging.info(
+            f'Initializing project in "{directory}" using template "{template}@{git_remote}"'
```

Comment:
Nitpicky: Maybe -> `{template}@{git_remote}:{git_commit}`?

Reply:
good catch let's just remove the git path here completely, since it may be None at this place :D also it's kinda trivial since you specified it in the CLI.